### ISSUE TITLE
void (sth) ?? (or ||) raises TS2873 and TS2869

### ISSUE THREAD

**Issue by amos-eye** [microsoft/TypeScript#60276]
### 🔎 Search Terms

TS2873, TS2869, void TS2873, void TS2869, This kind of expression is always falsy, Right operand of ?? is unreachable because the left operand is never nullish, void

### 🕗 Version & Regression Information

- This changed after 5.5.4 and can be found in 5.6.3 (Tried in online playground versions)

### 💻 Code

```ts
void 22 ?? true  // TS2873: This kind of expression is always falsy.
void 22 || true // TS2869: Right operand of ?? is unreachable because the left operand is never nullish.
```

### 🙁 Actual behavior

The void(sth) language construct causes a false positive rule to be triggered.

### 🙂 Expected behavior

This shouldn't happen or is it a warning?

### Additional information about the issue

The language often uses the construction: void (sth).
It is used to perform some action or a bunch of actions and return the last result in a chain of OR operators / nullish coalescing.
A common use case in the reduce function:

```js
data.reduce((acc, v) => void acc.push(...v) || acc, [])
```
